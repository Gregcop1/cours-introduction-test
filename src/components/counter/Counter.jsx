import React, { useState } from 'react';
import './counter.css';

export const Counter = ({ defaultValue = 0 }) => {
  const [value, changeValue] = useState(defaultValue);
  const decrement = () => changeValue(value - 1);
  const increment = () => changeValue(value + 1);
  const onChange = (e) => changeValue(+e.target.value);

  return (
    <div className="counter">
      <button className="btn-decrement" onClick={decrement}>-</button>
      <input type="text" value={value} onChange={onChange} />
      <button className="btn-increment" onClick={increment}>+</button>
    </div>
  )
};
