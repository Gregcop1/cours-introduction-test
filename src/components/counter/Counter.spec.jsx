import React from 'react';
import { Counter } from './Counter';
import { mount, shallow } from 'enzyme';

const setup = (props, takeScreenshot = false, renderMethod = shallow) => {
  const wrapper = renderMethod(<Counter {...props}/>);

  if (takeScreenshot) {
    expect(wrapper).toMatchSnapshot();
  }

  return wrapper;
};

describe('Counter', () => {
  it('should render the default view : Counter + 2 buttons', () => {
    setup(null, true);
  });

  it('should accept default value override', () => {
    setup({ defaultValue: 1 }, true);
  });

  it('should decrement value when a click on - button is triggered', () => {
    const wrapper = setup(null, false, mount);
    wrapper.find('.btn-decrement').simulate('click');
    expect(wrapper).toMatchSnapshot();
  });

  it('should increment value when a click on + button is triggered', () => {
    const wrapper = setup(null, false, mount);
    wrapper.find('.btn-increment').simulate('click');
    expect(wrapper).toMatchSnapshot();
  });
});
