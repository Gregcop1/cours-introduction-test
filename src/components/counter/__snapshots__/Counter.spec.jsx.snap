// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter should accept default value override 1`] = `
<div
  className="counter"
>
  <button
    className="btn-decrement"
    onClick={[Function]}
  >
    -
  </button>
  <input
    onChange={[Function]}
    type="text"
    value={1}
  />
  <button
    className="btn-increment"
    onClick={[Function]}
  >
    +
  </button>
</div>
`;

exports[`Counter should decrement value when a click on - button is triggered 1`] = `
<Counter>
  <div
    className="counter"
  >
    <button
      className="btn-decrement"
      onClick={[Function]}
    >
      -
    </button>
    <input
      onChange={[Function]}
      type="text"
      value={-1}
    />
    <button
      className="btn-increment"
      onClick={[Function]}
    >
      +
    </button>
  </div>
</Counter>
`;

exports[`Counter should increment value when a click on + button is triggered 1`] = `
<Counter>
  <div
    className="counter"
  >
    <button
      className="btn-decrement"
      onClick={[Function]}
    >
      -
    </button>
    <input
      onChange={[Function]}
      type="text"
      value={1}
    />
    <button
      className="btn-increment"
      onClick={[Function]}
    >
      +
    </button>
  </div>
</Counter>
`;

exports[`Counter should render the default view : Counter + 2 buttons 1`] = `
<div
  className="counter"
>
  <button
    className="btn-decrement"
    onClick={[Function]}
  >
    -
  </button>
  <input
    onChange={[Function]}
    type="text"
    value={0}
  />
  <button
    className="btn-increment"
    onClick={[Function]}
  >
    +
  </button>
</div>
`;
